src/form-validation.ts

```diff
@@ -0,0 +1,108 @@
+import { ValidationEngine } from './validation-engine';
+import {
+  ValidationSchema,
+  RecordValidationSchema,
+  FieldsValidationSchema,
+  FieldValidation,
+  ValidationResult,
+  FormValidationResult,
+} from './model';
+import { isFunction } from './helper';
+
+export class FormValidation {
+  private validationEngine: ValidationEngine;
+
+  constructor(validationSchema: ValidationSchema) {
+    this.validationEngine = new ValidationEngine();
+    this.setupValidationSchema(validationSchema);
+  }
+
+  // TODO: review parameters order, baseFormValidation validateField has fieldId, value, values
+  // the engine validateField has a different order, IMHO it should be always: fieldId, value, values
+  // since values is options.
+  public validateField(
+    fieldId: string,
+    value: any,
+    values?: any
+  ): Promise<ValidationResult> {
+    return this.validationEngine.validateField(values, fieldId, value);
+  }
+
+  public validateForm(values: any): Promise<FormValidationResult> {
+    return this.validationEngine.validateForm(values);
+  }
+
+  private setupValidationSchema(validationSchema: ValidationSchema) {
+    if (validationSchema && typeof validationSchema === 'object') {
+      const { record, fields } = validationSchema;
+
+      if (record && record instanceof Array) {
+        this.addRecordValidations(record);
+      }
+      if (fields && typeof fields === 'object') {
+        this.addAllFieldsValidations(fields);
+      }
+    } else {
+      console.error('ValidationSchema must be a valid object');
+    }
+  }
+
+  private addRecordValidations(
+    recordValidationSchemaCollection: RecordValidationSchema[]
+  ) {
+    recordValidationSchemaCollection.forEach(
+      (recordValidation: RecordValidationSchema) => {
+        this.addRecordValidation(recordValidation);
+      }
+    );
+  }
+
+  private addRecordValidation(recordValidation: RecordValidationSchema) {
+    if (isFunction(recordValidation)) {
+      this.validationEngine.addRecordValidation(recordValidation);
+    } else {
+      this.validationEngine.addRecordValidation(
+        recordValidation.validation,
+        recordValidation.message
+      );
+    }
+  }
+
+  private addAllFieldsValidations(fields: FieldsValidationSchema) {
+    for (const field in fields) {
+      this.addFieldValidations(field, fields[field]);
+    }
+  }
+
+  private addFieldValidations(
+    field: string,
+    fieldValidations: FieldValidation[]
+  ) {
+    if (fieldValidations instanceof Array) {
+      fieldValidations.forEach(fieldValidation => {
+        this.addFieldValidation(field, fieldValidation);
+      });
+    }
+  }
+
+  private addFieldValidation(field: string, fieldValidation: FieldValidation) {
+    if (isFunction(fieldValidation)) {
+      this.validationEngine.addFieldValidation(
```

Comment:
I think we have to rewrite `this.validationEngine.addFieldValidation` because we are split `fieldValidation` in 4 params to join it again

Reply:
You are right, I think it's job from BaseValidator to build this structure

Reply:
Done on refactor branch